sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -54,41 +53,40 @@ def get_token(
     "TokenCredential",
 ]
 
+T = TypeVar("T", str, Tuple[str, ...])
+
 
-class AzureKeyCredential:
+class AzureKeyCredential(Generic[T]):
     """Credential type used for authenticating to an Azure service.
     It provides the ability to update the key without creating a new client.
 
-    :param str key: The key used to authenticate to an Azure service
+    :param T key: The key used to authenticate to an Azure service
     :raises: TypeError
     """
 
-    def __init__(self, key: str) -> None:
-        if not isinstance(key, str):
-            raise TypeError("key must be a string.")
-        self._key = key
+    def __init__(self, key: T) -> None:
+        if not isinstance(key, (str, tuple)):
```

Comment:
Sorry, still find the tuple solution ugly.

What about *args, like we do for scope:
```python
def __init__(self, key, *keys)
```

Reply:
Then what is the update story?

For now, we require uses to update all of them as an atomic operation.

if we have def update(self, key, *keys), does that mean we accept arbitrary keys?

Reply:
And to make sure we are on the same page, if it is a one-key scenario, user can directly use AzureKeyCredential with 

cred = AzureKeyCredential(key=<str>)

For multi-key scenarios, user does not use AzureKeyCredential directly, instead, SDK developers should have a custom class which is a subclass of AzureKeyCredential and expose the custom credential to the users.

Could you give more details about in which case, the user experience is ugly?